lib/index.js

```diff
@@ -11,27 +11,48 @@ function ThisData (apiKey, options) {
   assert(apiKey, 'You must pass your ThisData api key');
 
   var options = options || {};
-  this.apiKey = apiKey;
-  this.host = options.host || 'https://api.thisdata.com';
   this.verbs = verbs;
   this.event = new Event(options);
+
+  this.client = request.defaults({
+    headers: {
+      'User-Agent': 'thisdata-node'
+    },
+    baseUrl: options.host || 'https://api.thisdata.com/v1/',
+    qs: {
+      api_key: apiKey
+    },
+    json: true
+  })
```

Comment:
what happens if there's an exception while making the request (something non-HTTP like a socket timeout or unknown host)? In the previous implementation those were caught but there it doesn't seem like they are. Just thinking that that might be an issue if ThisData has an API issue or a networking issue occurs at the consumers end.